src/Components/Server/src/Circuits/CircuitRegistry.cs

```diff
@@ -190,36 +211,45 @@ public virtual async Task<CircuitHost> ConnectAsync(string circuitId, IClientPro
 
         protected virtual (CircuitHost circuitHost, bool previouslyConnected) ConnectCore(string circuitId, IClientProxy clientProxy, string connectionId)
         {
-            if (ConnectedCircuits.TryGetValue(circuitId, out var circuitHost))
+            if (ConnectedCircuits.TryGetValue(circuitId, out var connectedCircuitHost))
             {
+                Log.ConnectingToActiveCircuit(_logger, circuitId, connectionId);
+
                 // The host is still active i.e. the server hasn't detected the client disconnect.
                 // However the client reconnected establishing a new connection.
-                circuitHost.Client.Transfer(clientProxy, connectionId);
-                return (circuitHost, true);
+                connectedCircuitHost.Client.Transfer(clientProxy, connectionId);
+                return (connectedCircuitHost, true);
             }
 
-            if (DisconnectedCircuits.TryGetValue(circuitId, out circuitHost))
+            if (DisconnectedCircuits.TryGetValue(circuitId, out DisconnectedCircuitEntry disconnectedEntry))
```

Comment:
Nit:

```suggestion
            if (DisconnectedCircuits.TryGetValue(circuitId, out var disconnectedEntry))
```

Reply:
This unfortunately doesn't work. `MemoryCache` returns an `object` instance by default (yes we're boxing these) and you have to specify the type to use the `TryGetValue<T>`  overload